pmml/pmml.py

```diff
@@ -139,80 +135,78 @@ def print_pmml(pmml_out):
-            print("\t</Header>", file=pmml_out)
-            # print out data dictionary entries for each column
-            print(
-                "\t<DataDictionary numberOfFields=\"%d\">" %
-                len(feature_names), file=pmml_out)
-            # not adding any interval definition, all values are currently
-            # valid
-            for feature in feature_names:
-                print(
-                    "\t\t<DataField name=\"" +
-                    feature +
-                    "\" optype=\"continuous\" dataType=\"double\"/>",
-                    file=pmml_out)
-            print("\t</DataDictionary>", file=pmml_out)
-            print("\t<MiningModel functionName=\"regression\">", file=pmml_out)
-            print("\t\t<MiningSchema>", file=pmml_out)
-            # list each feature name as a mining field, and treat all outliers
-            # as is, unless specified
-            for feature in feature_names:
-                print(
-                    "\t\t\t<MiningField name=\"%s\"/>" %
-                    (feature), file=pmml_out)
-            print("\t\t</MiningSchema>", file=pmml_out)
-            print(
-                "\t\t<Segmentation multipleModelMethod=\"sum\">",
-                file=pmml_out)
-            # read each array that contains pertinent information for the pmml
-            # these arrays will be used to recreate the traverse the decision
-            # tree
-            while model_content[line_no][:4] == 'Tree':
-                print("\t\t\t<Segment id=\"%d\">" % segment_id, file=pmml_out)
-                print("\t\t\t\t<True/>", file=pmml_out)
-                tree_no = model_content[line_no][5:]
-                num_leaves = int(get_value_string(model_content[line_no + 1]))
-                split_feature = get_array_ints(model_content[line_no + 2])
-                threshold = get_array_floats(model_content[line_no + 4])
-                decision_type = get_array_ints(model_content[line_no + 5])
-                left_child = get_array_ints(model_content[line_no + 6])
-                right_child = get_array_ints(model_content[line_no + 7])
-                leaf_parent = get_array_ints(model_content[line_no + 8])
-                leaf_value = get_array_floats(model_content[line_no + 9])
-                leaf_count = get_array_ints(model_content[line_no + 10])
-                internal_value = get_array_floats(model_content[line_no + 11])
-                internal_count = get_array_ints(model_content[line_no + 12])
-                unique_node_id = 0
-                print_pmml(pmml_out)
-                print("\t\t\t</Segment>", file=pmml_out)
-                line_no += 13
-                segment_id += 1
-
-            print("\t\t</Segmentation>", file=pmml_out)
-            print("\t</MiningModel>", file=pmml_out)
-            print("</PMML>", file=pmml_out)
-except Exception as ioex:
-    print(ioex)
+model_in = open(sys.argv[1])
+model_content = filter(
```

Comment:
Hi @wxchan instead of using `filter(None, model_in.readlines())`, I need to use 
`filter(None, model_in.read().splitlines())`, using the filter call with readlines wasn't working.